pkg/workloads/cortex/client/cortex/client.py

```diff
@@ -0,0 +1,124 @@
+import urllib.parse
+
+import dill
+import requests
+from requests.exceptions import HTTPError
+
+
+class Client(object):
+    def __init__(
+        self, aws_access_key_id, aws_secret_access_key, operator_url, workspace="./workspace"
+    ):
+        self.operator_url = operator_url
+        self.workspace = workspace
+        self.aws_access_key_id = aws_access_key_id
+        self.aws_secret_access_key = aws_secret_access_key
+        self.headers = {
+            "CortexAPIVersion": "master",
+            "Authorization": "CortexAWS {}|{}".format(
+                self.aws_access_key_id, self.aws_secret_access_key
+            ),
+        }
+
+        pathlib.Path(self.workspace).mkdir(parents=True, exist_ok=True)
+
+    def deploy(self, deployment_name, api_name, model_path, pre_inference, post_inference):
+        working_dir = os.path.join(self.workspace, deployment_name)
+        api_working_dir = os.path.join(working_dir, api_name)
+        pathlib.Path(api_working_dir).mkdir(parents=True, exist_ok=True)
+
+        reqs = subprocess.check_output([sys.executable, "-m", "pip", "freeze"])
+
+        with open(os.path.join(api_working_dir, "requirements.txt"), "w") as f:
+            f.writelines(reqs.decode())
+
+        with open(os.path.join(api_working_dir, "request_handler.pickle"), "wb") as f:
+            dill.dump(
+                {"pre_inference": pre_inference, "post_inference": post_inference}, f, recurse=True
+            )
+
+        cortex_config = [
+            {"kind": "deployment", "name": deployment_name},
+            {
+                "kind": "api",
+                "model": model_path,
+                "name": api_name,
+                "request_handler": "request_handler.pickle",
+            },
+        ]
+
+        cortex_yaml_path = os.path.join(working_dir, "cortex.yaml")
+        with open(cortex_yaml_path, "w") as f:
+            f.write(yaml.dump(cortex_config))
+
+        project_zip_path = os.path.join(working_dir, "project")
+        shutil.make_archive(project_zip_path, "zip", api_working_dir)
+        project_zip_path += ".zip"
+
+        queries = {"force": "false", "ignoreCache": "false"}
+
+        with open(cortex_yaml_path, "rb") as config, open(project_zip_path, "rb") as project:
+            files = {"cortex.yaml": config, "project.zip": project}
+            try:
+                resp = requests.post(
+                    urllib.parse.urljoin(self.operator_url, "deploy"),
+                    params=queries,
+                    files=files,
+                    headers=self.headers,
+                    verify=False,
+                )
+                resp.raise_for_status()
+            except HTTPError as err:
+                resp = err.response
+                if "error" in resp.json():
+                    raise Exception(resp.json()["error"]) from err
+                raise
+
+            if resp.status_code == 200:
+                return self.get_endpoint(deployment_name, api_name)
```

Comment:
Do you prefer this over the operator responding to the deploy() with the base URL and the context (which would eliminate the second request)? Do we intend users to also call `get_endpoint()`? If not we could merge it for now to clean up the API and avoid the extra request. Let me know if you want to discuss